guides/keras_tuner/failed_trials.py

```diff
@@ -0,0 +1,257 @@
+to 3 and only Trial 2, Trial 3, Trial 5, and Trial 6 fail, the search would not
+be terminated since the failed trials are not consecutive. The default value of
+`max_consecutive_failed_trials` is 3.
+
+The following code shows how these two arguments work in action.
+
+* We define a search space with 2 hyperparameters for the number of units in the
+  2 dense layers.
+* When their product is larger than 800, we raise a `ValueError` for the model
+  too large.
+
+"""
+
+
+def build_model(hp):
+    # Define the 2 hyperparameters for the units in dense layers
+    units_1 = hp.Int("units_1", 10, 40, step=10)
+    units_2 = hp.Int("units_2", 10, 30, step=10)
+
+    # Define the model
+    model = keras.Sequential(
+        [
+            layers.Dense(units=units_1, input_shape=(20,)),
+            layers.Dense(units=units_2),
+            layers.Dense(units=1),
+        ]
+    )
+    model.compile(loss="mse")
+
+    # Raise an error when the model is too large
+    n_params = model.count_params()
+    if n_params > 1200:
+        raise ValueError(f"Model too large! It contains {n_params} params.")
+    return model
+
+
+"""
+We set up the tuner as follows.
+
+* We set `max_retries_per_trial=3`.
+* We set `max_consecutive_failed_trials=8`.
+* We use `GridSearch` to enumerate all hyperparameter value combinations.
+
+"""
+
+tuner = keras_tuner.GridSearch(
+    hypermodel=build_model,
+    objective="val_loss",
+    overwrite=True,
+    max_retries_per_trial=3,
+    max_consecutive_failed_trials=8,
+)
+
+# Use random data to train the model.
+tuner.search(
+    x=np.random.rand(100, 20),
+    y=np.random.rand(100, 1),
+    validation_data=(
+        np.random.rand(100, 20),
+        np.random.rand(100, 1),
+    ),
+    epochs=10,
+)
+
+# Print the results.
+tuner.results_summary()
+
+"""
+## Mark a trial as failed
+
+When the model is too large, we do not need to retry it. No matter how many
+times we try with the same hyperparameters, it is always too large.
+
+We can set `max_retries_per_trial=0` to do it. However, it will not retry no
+matter what errors are raised while we may still want to retry for other
+unexpected errors. Is there a way to better handle this situation?
+
+We can raise the `FailedTrialError` to skip the retries. Whenever, this error is
+raised, the trial would not be retried. The retries will still run when other
+errors occur. An example is shown as follows.
+"""
+
+
+def build_model(hp):
+    # Define the 2 hyperparameters for the units in dense layers
+    units_1 = hp.Int("units_1", 10, 40, step=10)
+    units_2 = hp.Int("units_2", 10, 30, step=10)
+
+    # Define the model
+    model = keras.Sequential(
+        [
+            layers.Dense(units=units_1, input_shape=(20,)),
+            layers.Dense(units=units_2),
+            layers.Dense(units=1),
+        ]
+    )
+    model.compile(loss="mse")
+
+    # Raise an error when the model is too large
+    n_params = model.count_params()
```

Comment:
Please `num_` as the number prefix, for consistency.

Reply:
Done.